code/Model/VirtualPage.php

```diff
@@ -368,6 +369,22 @@ public function CMSTreeClasses()
         return parent::CMSTreeClasses() . ' VirtualPage-' . $this->CopyContentFrom()->ClassName;
     }
 
+    /**
+     * Use the target page's class name for fetching templates - as we need to take on its appearance
+     *
+     * @param string $suffix
+     * @return array
+     */
+    public function getViewerTemplates($suffix = '')
+    {
+        $copy = $this->CopyContentFrom();
+        if ($copy && $copy->exists()) {
+            return SSViewer::get_templates_by_class(get_class($copy), $suffix, SiteTree::class);
```

Comment:
`$copy->getViewerTemplates($suffix);` would be more respectful to custom implementations also.